deluge/tests/test_config.py

```diff
@@ -83,6 +84,55 @@ def test_set_get_item_none(self):
 
         config._save_timer.cancel()
 
+    @pytest_twisted.inlineCallbacks
+    def test_on_changed_callback(self):
+        def changed_cb(key, value):
+            nonlocal called
+            called = True
+            assert key == 'foo'
+            assert value == 1
+
+        def check_called():
+            assert called
+
+        called = False
+        config = Config('test.conf', config_dir=self.config_dir)
+        config.register_change_callback(changed_cb)
+        config['foo'] = 1
+        assert config['foo'] == 1
+        yield task.deferLater(reactor, 0.5, check_called)
```

Comment:
I think we could use `unittest.mock.Mock` to simplify these callbacks a bit, as well as eliminate the problem of having to wait longer than needed for them.
```python
def mock_callback(timeout=0.5):
    """Returns a `Mock` object which can be registered as a callback to test against.

    If callback was not called within `timeout` seconds, it will raise a TimeoutError.
    The returned Mock instance will have a `deferred` attribute which will complete when the callback has been called.
    """
    deferred = defer.Deferred()
    deferred.addTimeout(timeout, reactor)
    mock = unittest.mock.Mock(side_effect=deferred.callback)
    mock.deferred = deferred
    return mock

cb = mock_callback()
config.register_change_callback(cb)
config['foo'] = 1
await cb.deferred
cb.assert_called_once_with('foo', 1)
```

If we need to test registering callbacks like this in any other tests, maybe the mock_callback functions goes somewhere common to import.

Reply:
I have created a fixture from this code and changed the tests